### PR TITLE
move `TfGraphDefWrapper` and  `TfGraphDefProducer` from `RecoTracker/FinalTrackSelectors` to `PhysicsTools/TensorFlow`

### LINKED ISSUE #37958: Find a better home for RecoTracker.FinalTrackSelectors._tfGraphDefProducer
The ESProducer [TfGraphDefProducer](https://github.com/cms-sw/cmssw/blob/726783246ef4bb47cc730c31e12d7d31bf355c91/RecoTracker/FinalTrackSelectors/plugins/TfGraphDefProducer.cc) was introduced for the TrackDnn, but it's in fact completely generic.

TICL is now also using this module to load the TF graph through the EventSetup, and recently we were informed that Pixel CPE is also interested in using it. We are now in a situation where several groups will be importing something from tracking that doesn't have anything to do with tracking per se.

Therefore, it would be good to move `TfGraphDefProducer` from `RecoTracker/FinalTrackSelectors` to a new place, e.g. `PhysicsTools/TensorFlow`.

Then also the python config for TrackDnn TICL needs to be modified accordingly (they can be found with `git grep tfGraphDefProducer`).

### PR DESCRIPTION
**Pull request by mmusich**
resolves https://github.com/cms-sw/cmssw/issues/37958

#### PR description:

As per https://github.com/cms-sw/cmssw/issues/37958#issue-1237108189.
Also moved `TfGraphRecord` from `TrackingTools/Records` to `PhysicsTools/TensorFlow` in commit https://github.com/cms-sw/cmssw/pull/37968/commits/ae0c89c43bb3110521519bf1d19009457dbe0ea

#### PR validation:

`cmssw` compiles

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

File: PhysicsTools/TensorFlow/plugins/TfGraphDefProducer.cc [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 #include "FWCore/Framework/interface/ESHandle.h"
 #include "TrackingTools/Records/interface/TfGraphRecord.h"
```
Comment (mmusich) [inline PhysicsTools/TensorFlow/plugins/TfGraphDefProducer.cc:23]: should this file be migrated as well? (should be generic enough)

https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/TrackingTools/Records/interface/TfGraphRecord.h#L1-L22

(despite the description).
If so what's a good package to move it?

Comment (jpata) [reply PhysicsTools/TensorFlow/plugins/TfGraphDefProducer.cc:23]: good point. by analogy with MVAComputer, is there any objection to putting the TfGraphRecord in PhysicsTools/TensorFlow?

```
PhysicsTools/MVAComputer/interface/MVAComputerRecord.h:#include "FWCore/Framework/interface/EventSetupRecordImplementation.h"
```

**Comment by jpata** [comment]
@riga @yongbinfeng can you please review this from the ML side? Potentially some info on the TfGraphDef could be in the ML docs as well.

**Comment by jpata** [comment]
@cmsbuild please test

File: PhysicsTools/TensorFlow/plugins/TfGraphDefProducer.cc (hunk 1 above)
Comment (mmusich) [reply PhysicsTools/TensorFlow/plugins/TfGraphDefProducer.cc:23]: so I'll go ahead with this change as well, but I'll wait for tomorrow to see if there are further comments from the ML people.

Comment (mmusich) [reply PhysicsTools/TensorFlow/plugins/TfGraphDefProducer.cc]: done at [ae0c89c](https://github.com/cms-sw/cmssw/pull/37968/commits/ae0c89c43bb3110521519bf1d19009457dbe0eac)

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by jpata** [comment]
+reconstruction
- technical, addresses https://github.com/cms-sw/cmssw/issues/37958 (code cleanup for the future)

**Comment by mmusich** [comment]
@cms-sw/upgrade-l2 please check and sign.
Thank you

**Comment by srimanob** [comment]
+Upgrade

This consider to be technical PR follow the suggestion in https://github.com/cms-sw/cmssw/issues/37958

**Comment by perrotta** [comment]
+1

**Pull request merged**